source/include/azure_iot_config_defaults.h

```diff
@@ -0,0 +1,124 @@
+/* Copyright (c) Microsoft Corporation. All rights reserved. */
+/* SPDX-License-Identifier: MIT */
+
+#ifndef AZURE_IOT_CONFIG_DEFAULTS_H
+#define AZURE_IOT_CONFIG_DEFAULTS_H
+
+#ifndef azureiotDEFAULT_TOKEN_TIMEOUT_IN_SEC
```

Comment:
Even though these are somewhat self-documenting, I feel some additional doxygen would help here so there's not ambiguity.